worker/changestream/eventmultiplexer/notifyworker.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventmultiplexer
+
+import (
+	"sync/atomic"
+
+	"github.com/juju/juju/core/changestream"
+	"gopkg.in/tomb.v2"
+)
+
+type Signaller interface {
+	Signal(changes []changestream.ChangeEvent)
+}
+
+type notifyWorker struct {
+	tomb tomb.Tomb
+
+	notary      chan struct{}
+	numNotaries int64
+}
+
+func newNotifyWorker(size int64) *notifyWorker {
+	w := &notifyWorker{
+		numNotaries: size,
+		notary:      make(chan struct{}),
+	}
+	w.tomb.Go(w.loop)
+	return w
+}
+
+func (s *notifyWorker) Notify(sub *subscription, changes []changestream.ChangeEvent) {
```

Comment:
Obviously Notify is not safe to call more than `numNotaries` times. The call tomb.Go to will likely panic.

Reply:
Actually this seems problematic because before the call to Notify, we add the notifyWorker to the EventMultiplexer's catacomb, which means the tomb could have died by the time one of these Notify calls are made, causing a panic.